core/src/utils/wrapper_opaque.rs

```diff
@@ -113,8 +116,6 @@ impl<T, R: TypeResolver> Visitor for WrapperKeepOpaqueVisitor<T, R> {
     ) -> Result<Self::Value<'scale, 'info>, Self::Error> {
         use scale_decode::error::{Error, ErrorKind};
 
```

Comment:
I'd like to keep that TODO unless there's an issue for it; just a remindfer for future us to add the extra check here in the future :)